viennacl/src/main/scala/org/apache/mahout/viennacl/opencl/javacpp/DenseColumnMatrix.scala

```diff
@@ -36,12 +36,24 @@ import org.bytedeco.javacpp.annotation._
 @Name(Array("viennacl::matrix<double,viennacl::column_major>"))
 final class DenseColumnMatrix(initDefault:Boolean = true) extends MatrixBase {
 
-  def this(nrow: Int, ncol: Int, ctx: Context = new Context()) {
+  def this(nrow: Int, ncol: Int) {
+    this(false)
+    allocate(nrow, ncol, new Context())
+  }
+
+  def this(nrow: Int, ncol: Int, ctx: Context) {
     this(false)
     allocate(nrow, ncol, ctx)
   }
 
-  def this(data: DoublePointer, nrow: Int, ncol: Int, ctx: Context = new Context(Context.MAIN_MEMORY)) {
```

Comment:
@andrewpalumbo thanks for the review

If you look at line 49 and 56 below- this is what replaces.  Line 49 lets you call the constructor with out a default context (same as what happened here) line 56 lets you explicitly pass a context. 

Re: Scala 2.11.2, I made a comment on MAHOUT-1975.  We probably need to figure out where the dependency comes from and try to work around it. 

Will also dig in on my end- viennacl always fails for me bc I don't have strong enough GPUs- but results should be same for ViennaCL-OMP.